src/oatpp/core/provider/Pool.hpp

```diff
@@ -223,22 +223,24 @@ class PoolTemplate : public oatpp::base::Countable, public async::CoroutineWaitL
   {
     /* "new" is called directly to keep constructor private */
     auto ptr = std::shared_ptr<PoolTemplate>(new PoolTemplate(provider, maxResources, maxResourceTTL.count()));
-    ptr->startCleanupTask();
+    ptr->startCleanupTask(ptr);
```

Comment:
What about get() and getAsync(). 
As far as I can see the get passed a shared_ptr to this so that the AcquisitionProxy keeps the pool alive.
I don't think it is desirable to call both functions with a shared_ptr pointing to another object than this.

Should they be also static? I am not sure how much current users of oat++ will be affected by such a change.
Maybe they would have to make changes to their code to upgrade to a newer version if this change is made.

Reply:
Yes, we can make  `get(const std::shared_ptr<PoolTemplate>& _this)` and `getAsync(const std::shared_ptr<PoolTemplate>& _this)` static and protected.
Users won't be affected since they are not supposed to use those functions directly.
Instead user should use `get()` and `getAsync()`.